src/rgw/rgw_rest_s3.cc

```diff
@@ -3960,6 +3960,37 @@ int RGWInitMultipart_ObjStore_S3::get_params(optional_yield y)
 
   policy = s3policy;
 
+  //handle object lock
+  auto obj_lock_mode_str = s->info.env->get("HTTP_X_AMZ_OBJECT_LOCK_MODE");
+  auto obj_lock_date_str = s->info.env->get("HTTP_X_AMZ_OBJECT_LOCK_RETAIN_UNTIL_DATE");
+  auto obj_legal_hold_str = s->info.env->get("HTTP_X_AMZ_OBJECT_LOCK_LEGAL_HOLD");
```

Comment:
this looks mostly copied from PutObj/CopyObj, except with the improvement to use std::optional. it would be nice if the ops shared a helper function instead of duplicating the code again